src/meshes/triangle_mesh.template.cc

```diff
@@ -36,6 +36,177 @@
 
 namespace oomph
 {
+  //======================================================================
+  /// Build mesh from the structure and geometry of another mesh.
+  /// A lot of this code is repeated in the construction of the mesh from
+  /// the scaffold mesh.
+  //======================================================================
+  template<class ELEMENT>
+  template<class ORIG_ELEMENT>
+  void TriangleMesh<ELEMENT>::build_from_another_mesh(
+    TriangleMesh<ORIG_ELEMENT>* orig_mesh_pt)
+  {
```

Comment:
Didn't we need a warning that this won't work if run in parallel (if true!). Maybe I mis-remembered

Reply:
It currently passes the MPI tests on Github but probably won't work if the mesh is segregated etc.
What should I be testing for? A blank MPI test seems a bit excessive.

Reply:
Ah, yes, I mean distributed. I think meshes have a boolean member function that says if they're distributed, so if you suspect that this functionality won't work in such cases (I agree) we should bail out as a precaution.